streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/KTableRepartitionerProcessorSupplier.java

```diff
@@ -0,0 +1,95 @@
+    private final Serializer<V> valueSerializer;
+
+    public KTableRepartitionerProcessorSupplier(final ValueMapper<V, KO> extractor,
+                                                final Serializer<V> valueSerializer) {
+        this.mapper = extractor;
+        this.valueSerializer = valueSerializer;
+    }
+
+    @Override
+    public Processor<K, Change<V>> get() {
+        return new UnbindChangeProcessor();
+    }
+
+    private class UnbindChangeProcessor extends AbstractProcessor<K, Change<V>> {
+        
+        @Override
+        public void init(final ProcessorContext context) {
+            super.init(context);
+        }
+
+        @Override
+        public void process(final K key, final Change<V> change) {
+            long[] nullHash = Murmur3.hash128(new byte[]{});
+            long[] currentHash = (change.newValue == null ?
+                    Murmur3.hash128(new byte[]{}):
+                    Murmur3.hash128(valueSerializer.serialize(context().topic(), change.newValue)));
+
+            if (change.oldValue != null) {
+                final KO oldForeignKey = mapper.apply(change.oldValue);
+                final CombinedKey<KO, K> combinedOldKey = new CombinedKey<>(oldForeignKey, key);
+                if (change.newValue != null) {
+                    final KO extractedNewForeignKey = mapper.apply(change.newValue);
+                    final CombinedKey<KO, K> combinedNewKey = new CombinedKey<>(extractedNewForeignKey, key);
+
+                    //Requires equal to be defined...
+                    if (oldForeignKey.equals(extractedNewForeignKey)) {
+                        //Same foreign key. Just propagate onwards.
+                        context().forward(combinedNewKey, new SubscriptionWrapper(currentHash, true));
+                    } else {
+                        //Different Foreign Key - delete the old key value and propagate the new one.
+                        //Note that we indicate that we don't want to propagate the delete to the join output.
+                        //The downstream processor to delete it from the local state store, but not propagate it.
+                        context().forward(combinedOldKey, new SubscriptionWrapper(nullHash, false));
+                        context().forward(combinedNewKey, new SubscriptionWrapper(currentHash, true));
+                    }
+                } else {
+                    //A propagatable delete. Set hash to null instead of using the null hash code.
+                    context().forward(combinedOldKey, new SubscriptionWrapper(nullHash, true));
+                }
+            } else if (change.newValue != null) {
+                final KO extractedForeignKeyValue = mapper.apply(change.newValue);
+                final CombinedKey<KO, K> newCombinedKeyValue = new CombinedKey<>(extractedForeignKeyValue, key);
+                context().forward(newCombinedKeyValue, new SubscriptionWrapper(currentHash, true));
```

Comment:
Yep, you're correct. I believe this is an artifact left over from the initial design, when I used to send (CombinedKey<KO,K>, V) as the event, and it was either put the FK in the Key or in the Value. I'll adjust the code accordingly.

Reply:
Okay, while I took it out from over-the-wire, but we still need the Serde for the `prefixScanStoreBuilder`, since CombinedKeys are stored to the changelog topic. The end result is that the only thing that is removed is the custom partitioner, which I like as I we no longer rely on a copied + pasted + slightly-edited partitioner.